scripts/build/tests.js

```diff
@@ -167,23 +156,26 @@ exports.cleanTestDirs = cleanTestDirs;
  * @param {boolean} [keepFailed]
  * @param {number | undefined} [shards]
  * @param {number | undefined} [shardId]
+ * @param {boolean} [color]
  */
-function writeTestConfigFile(tests, runners, light, taskConfigsFolder, workerCount, stackTraceLimit, timeout, keepFailed, shards, shardId) {
+function writeTestConfigFile(reporter, reporterOption, tests, runners, light, taskConfigsFolder, workerCount, stackTraceLimit, timeout, keepFailed, shards, shardId, color) {
     const testConfigContents = JSON.stringify({
-        test: tests ? [tests] : undefined,
-        runners: runners ? runners.split(",") : undefined,
+        reporter,
+        reporterOption,
+        grep: tests || undefined, // tests is a string interpreted as a regexp
+        runners: runners ? runners.split(/ *, */) : undefined,

```

Comment:
This won't catch `--runners " compiler,conformance "`. If we're going to change this, it might be better to do the following instead:
```suggestion
        runners: runners ? runners.split(/,/).map(runner => runner.trim()) : undefined,
```

Reply:
Yeah, it's easy to add a `.trim` and possibly a `.filter` to deal with `",foo,bar,"`, but I just did the simple thing thinking that it's easy to make a `"foo, bar"` mistake, but the others are unlikely?  (Will fix as wanted if it's not dumped...)